test/gong/connector.go

```diff
@@ -2,55 +2,50 @@ package gong
 
 import (
 	"context"
-	"net/http"
-
-	"github.com/amp-labs/connectors/common/scanning"
+	"github.com/amp-labs/connectors/common/scanning/credscanning"
+	"github.com/amp-labs/connectors/providers"
 	"github.com/amp-labs/connectors/providers/gong"
-	testUtils "github.com/amp-labs/connectors/test/utils"
-	"github.com/amp-labs/connectors/utils"
+	"github.com/amp-labs/connectors/test/utils"
+	"golang.org/x/oauth2"
+	"net/http"
 )
 
-func GetGongConnector(ctx context.Context, filePath string) *gong.Connector {
-	registry := scanning.NewRegistry()
-
-	readers := []scanning.Reader{
-		&scanning.JSONReader{
```

Comment:
Where is this mapping living now? Or is it redundant now?

Reply:
For most part it is redundant. We shouldn't specify what fields are to be read from the file. The provider catalog has all the information (at least it should in final catalog version, ex: require workspace flag is yet to be used by all types of connectors, we explicitly say we need this, in future it should be implied as well).

In here we imply fields using `providerInfo`: https://github.com/amp-labs/connectors/blob/3c4deeb3cf1f01e97b9f836b90bdb7d96c7c255e/common/scanning/credscanning/provider.go#L62 (and flags, which are temporary as I said).

Then for every field create reader: https://github.com/amp-labs/connectors/blob/3c4deeb3cf1f01e97b9f836b90bdb7d96c7c255e/common/scanning/credscanning/provider.go#L73

And finally validate required fields. Every field key and value will be cached in the map. https://github.com/amp-labs/connectors/blob/3c4deeb3cf1f01e97b9f836b90bdb7d96c7c255e/common/scanning/credscanning/provider.go#L87

When making `ProviderCredentials.Get` call it will give you that string value. https://github.com/amp-labs/connectors/blob/3c4deeb3cf1f01e97b9f836b90bdb7d96c7c255e/common/scanning/credscanning/provider.go#L135